### PR TITLE
Refactor ``_is_type_checking_import`` of ``unused-import``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

https://github.com/PyCQA/pylint/pull/5316#discussion_r757791192 and adds a changelog for that PR.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -22,6 +22,9 @@ Release date: TBA
 ..
   Put bug fixes that should not wait for a new minor version here
 
+* Fixed a false positive for ``unused-import`` where everything
+  was not analysed properly inside typing guards
```
Comment (cdce8p) [inline ChangeLog:26]:
```suggestion
  was not analyzed properly inside typing guards.
```
I usually try to end changelog entries with a full stop.

Comment (DanielNoord) [reply ChangeLog:26]: 👍

Comment (DanielNoord) [reply ChangeLog]: I copied the entry from #5397 to avoid a conflict for @Pierre-Sassoulas but he will have to update his PR

**Review by cdce8p — APPROVED** [review]
Looks good. Thanks @DanielNoord 🐬
Github is having some issue at the moment, so I think it's fine even if the `CodeQL` job fails.

**Pull request merged**